test/method/testddec.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (c) 2020, the cclib development team
+#
+# This file is part of cclib (http://cclib.github.io) and is distributed under
+# the terms of the BSD 3-Clause License.
+
+"""Test the DDEC6 in cclib"""
+
+from __future__ import print_function
+
+import sys
+import os
+import logging
+import unittest
+
+import numpy
+
+from cclib.method import DDEC6, volume
+from cclib.parser import Psi4
+from cclib.method.calculationmethod import MissingAttributeError
+
+from numpy.testing import assert_allclose
+
+from ..test_data import getdatafile
+
+
+class DDEC6Test(unittest.TestCase):
+    """DDEC6 method tests."""
+
+    def setUp(self):
+        super(DDEC6Test, self).setUp()
+        self.parse()
+
+    def parse(self):
+        self.data, self.logfile = getdatafile(Psi4, "basicPsi4-1.2.1", ["water_mp2.out"])
+        self.volume = volume.Volume((-4, -4, -4), (4, 4, 4), (0.2, 0.2, 0.2))
+
+    def testmissingrequiredattributes(self):
+        """Is an error raised when required attributes are missing?"""
+        for missing_attribute in DDEC6.required_attrs:
+            self.parse()
+            delattr(self.data, missing_attribute)
+            with self.assertRaises(MissingAttributeError):
+                trialBader = DDEC6(self.data, self.volume)
+
+    def test_proatom_read(self):
+        """Are proatom densities imported correctly?"""
+
+        self.data, logfile = getdatafile(Psi4, "basicPsi4-1.2.1", ["water_mp2.out"])
+        self.volume = volume.Volume((-4, -4, -4), (4, 4, 4), (0.2, 0.2, 0.2))
```

Comment:
`self.parse()` here? Unless you were planning to do this on an even coarser grid since its not used in this test?

Reply:
Yes -- thanks for catching this! I've made the necessary changes.